pkg/clusters/addons/kong/builder.go

```diff
@@ -36,6 +41,34 @@ func (b *Builder) WithPostgreSQL() *Builder {
 	return b
 }
 
+// WithEnterprise deploying kong enterpise
+func (b *Builder) WithEnterprise() *Builder {
+	b.enterprise = true
+	b.repo = DefaultEnterpriseImageRepo
+	b.tag = DefaultEnterpriseImageTag
+	b.license = KongEnterpriseTestLicense
+	b.kongAdminPassword = EnterpriseKongAdminDefaultPWD
+	return b
+}
+
+// WithImage specify docker image repo and tag
+func (b *Builder) WithImage(repo, tag string) *Builder {
+	b.repo = repo
+	b.tag = tag
+	return b
+}
+
+// WithLicense specify license secret name
+func (b *Builder) WithLicense(license string) *Builder {
+	b.license = license
+	return b
+}
+
+func (b *Builder) WithKongAdminPassword(password string) *Builder {
+	b.kongAdminPassword = password
+	return b
+}
```

Comment:
Related to an above comment it seems like right now we only intend this feature to be used when enterprise is configured, so we should improve the naming:

```suggestion
func (b *Builder) WithEnterpriseSuperuserPassword(password string) *Builder {
	b.kongAdminPassword = password
	return b
}
```

Reply:
I'm going to pass on this one for now, since @ccfishk has shown a strong preference for the "KongAdminPassword" naming.